.github/workflows/ci-e2e-tests.yml

```diff
@@ -0,0 +1,178 @@
+name: E2E tests
+
+permissions:
+  id-token: write
+  contents: read
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - "v*.*.*"
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  e2e_tests:
+    name: E2E tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Git checkout
+        uses: actions/checkout@v3
+      
+      - name: Install Nix
+        uses: cachix/install-nix-action@v27
+        # relax sandbox due impure kontrol-service tests during build
+        with:
+          extra_nix_config: |
+            sandbox = relaxed
+
+      - name: Magic cache
+        uses: DeterminateSystems/magic-nix-cache-action@v7
+
+      - name: Start Kardinal Kontrol service and Postgres 
+        run: |
+          docker version
+          docker compose -f ci/docker-compose.yml up -d
+          docker ps
+
+      - name: Wait for docker network to be ready
+        run: sleep 10s
+        shell: bash
+
+      - name: Start minikube
+        uses: manusa/actions-setup-minikube@v2.11.0
+        id: minikube
+        with:
+          minikube version: "v1.33.0"
+          kubernetes version: "v1.30.0"
+          driver: docker
+          start args: --embed-certs --addons=ingress,metrics-server
+      
+      - name: Install Istio
+        run: |
+          ISTIO_VERSION="1.23.0"
+          echo "Installing Istio ..."
+          mkdir istio_tmp
+          pushd istio_tmp >/dev/null
+              curl -L https://istio.io/downloadIstio | ISTIO_VERSION=${ISTIO_VERSION} sh -
+              cd istio-${ISTIO_VERSION}
+              export PATH=$PWD/bin:$PATH
+              istioctl install --set profile=demo -y
+          popd
+ 
+      - name: Build Kardinal manager and load image
+        run: |
+          eval $(minikube docker-env)
+          docker load < $(nix build ./#containers.x86_64-linux.kardinal-manager.amd64 --no-link --print-out-paths)
+          docker tag kurtosistech/kardinal-manager:latest-amd64 kurtosistech/kardinal-manager:latest     
+          docker image ls
+
+      - name: Build Kardinal CLI
+        run: |
+          result=$(nix build ./#kardinal-cli --no-link --print-out-paths)
+          path=$(find $result -name 'kardinal.cli*' -type f | head -n 1)
+          binout="/tmp/kardinal-cli"
+          ln -s $path $binout
+          if /tmp/kardinal-cli | grep -q "Kardinal CLI"; then exit 0; else exit 1; fi
+
+      - name: Retrieve the tenant UUID
+        id: tenant
+        run: |
+          tenant_id=$(/tmp/kardinal-cli tenant show)
+          echo "id=${tenant_id}" >> "$GITHUB_OUTPUT"
+
+      - name: Deploy Kardinal manager
+        run: |
+          KARDINAL_CLI_DEV_MODE=TRUE /tmp/kardinal-cli manager deploy local-minikube
```

Comment:
I think you could use a constant for "/tmp/kardinal-cli" because it's used in several steps

Reply:
I tried with env variables and output variables but it is too verbose so i am keeping it that way for now.